### ISSUE TITLE
Latest version of concurrent-ruby has dropped support for TimerTask timeouts

### ISSUE THREAD

**Issue by robbavey** [elastic/logstash#13956]
concurrent-ruby `1.10` has [dropped](https://github.com/ruby-concurrency/concurrent-ruby/pull/926) support for TimerTask timeouts, due to issues with leaking threads. 

These are still used in two different places in the Logstash codebase - 
https://github.com/elastic/logstash/blob/55e7a2641616b6c94e1390a6a921b54650ecd5a3/x-pack/lib/monitoring/inputs/metrics.rb#L61

https://github.com/elastic/logstash/blob/55e7a2641616b6c94e1390a6a921b54650ecd5a3/logstash-core/lib/logstash/instrument/periodic_poller/base.rb#L85

**Logstash information**:

Please include the following information:

1. Logstash version (e.g. `bin/logstash --version`)
`main` (`8.3.0-SNAPSHOT`)

**Description of the problem including expected versus actual behavior**:

Starting up logstash, the following is output to `stdout`

```
TimerTask timeouts are now ignored as these were not able to be implemented correctly
TimerTask timeouts are now ignored as these were not able to be implemented correctly
TimerTask timeouts are now ignored as these were not able to be implemented correctly
TimerTask timeouts are now ignored as these were not able to be implemented correctly
```

**Steps to reproduce**:

`bin/logstash -e 'input {stdin {}}'`

**Provide logs (if relevant)**:

**Comment by wallrik** [comment]
This error unfortunately shipped to production in version `8.2.0`.

**Comment by knightian** [comment]
Yep noticed it on logstash 8.2 just now

**Comment by robbavey** [comment]
Closed by #14113
